integration/framework/scenario.go

```diff
@@ -1,41 +1,68 @@
-package framework
+package e2e
 
 import (
 	"fmt"
+	"io/ioutil"
 	"os"
+	"path/filepath"
 	"strings"
 
 	"github.com/pkg/errors"
 )
 
 const (
-	MinioAccessKey = "cortex"
-	MinioSecretKey = "supersecret"
-	NetworkName    = "cortex-integration-test"
+	NetworkName        = "e2e-test"
+	ContainerSharedDir = "/shared"
 )
 
 type Scenario struct {
 	services []*Service
+
+	dir string
 }
 
 func NewScenario() (*Scenario, error) {
 	s := &Scenario{
 		services: []*Service{},
 	}
 
+	if os.Getenv("E2E_INTEGRATION_TEST_DIR") != "" {
+		s.dir = os.Getenv("E2E_INTEGRATION_TEST_DIR")
+		if err := os.MkdirAll(s.dir, os.ModePerm); err != nil {
+			return nil, err
+		}
+
+	} else {
+		tmpDir, err := ioutil.TempDir(".", "e2e_integration_test")
+		if err != nil {
+			return nil, err
+		}
+		s.dir, err = filepath.Abs(tmpDir)
+		if err != nil {
+			_ = os.RemoveAll(tmpDir)
+			return nil, err
+		}
+	}
+
 	// Force a shutdown in order to cleanup from a spurious situation in case
 	// the previous tests run didn't cleanup correctly
-	s.Shutdown()
+	s.shutdown()
 
 	// Setup the docker network
 	if out, err := RunCommandAndGetOutput("docker", "network", "create", NetworkName); err != nil {
 		fmt.Println(string(out))
+		_ = os.RemoveAll(s.dir)
```

Comment:
Is this require, given we now call `s.Close()` even in case `NewScenario()` returns error?

Reply:
> ow call s.Close() even in case NewScenario() returns error?

We don't as scenario is nil.

Reply:
Right, then we can call `s.clean()` here, right?

Reply:
:+1: